recipe/meta.yaml

```diff
@@ -8,53 +7,79 @@ package:
 source:
   fn: menuinst-{{ version }}.tar.gz
   url: https://github.com/conda/menuinst/archive/{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  sha256: 0ca844379c0d904cc1879f574c4d4320e89290e63e2f61e2996aebdd579d539e
 
 build:
-  number: 1
-  skip: True  # [not win]
+  number: 0
+  skip: true  # [py<38]
+  script_env:
+    - SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
   script:
-    - {{ PYTHON }} -m pip install . -vv
-    - copy "%SRC_DIR%\\cwp.py" "%PREFIX%\\"
-  entry_points:
-    - menuinst = menuinst.main:main
-  skip_compile_pyc:
-    - cwp.py
+    # Apparently these files make the post-build linkage analysis crash
+    # and we should not need them on Windows
+    - del /q "{{ SRC_DIR }}\\menuinst\\data\\osx_launcher_*"     # [win]
+    - del /q "{{ SRC_DIR }}\\menuinst\\data\\appkit_launcher_*"  # [win]
+    # Delete in Linux too because we don't need them there
+    - rm -f "{{ SRC_DIR }}/menuinst/data/osx_launcher_*"     # [linux]
+    - rm -f "{{ SRC_DIR }}/menuinst/data/appkit_launcher_*"  # [linux]
+    - {{ PYTHON }} -m pip install . -vv --no-build-isolation --no-deps
+    # menuinst v1 expects cwp.py in PREFIX; backwards compat fix
+    - copy "%SP_DIR%\\menuinst\\_legacy\cwp.py" "%PREFIX%\\"  # [win]
+  skip_compile_pyc:  # [win]
+    - cwp.py  # [win]
+  missing_dso_whitelist:
+    - '**/libswift*'  # [osx]
```

Comment:
@jaimergp Just to be sure, I took a guess here since I think these are always available on any macOS (considering that Swift originates from Apple and it's part of their ecosystem), but it could be great if you could confirm that this is the case.

Reply:
Yes, check [this PR](https://github.com/conda/menuinst/pull/160) where I am documenting the Swift deal.

The launcher that requires Swift is only needed for a small subset of the menuinst functionality. Swift is present in 10.14.4 and above. For earlier versions, Apple distributes the runtime.

So all in all we can add them to the DSO allow-list.

Reply:
Thanks. So I guess we should add `__osx>=10.14.4` but I don't know if it would break our ecosystem. @chenghlee @jezdez any idea?

Reply:
That would force conda too. It's fine to leave it as it is. Packages creating shortcut can add the __osx requirement if needed.